packages/eas-cli/src/commands/device/delete.ts

```diff
@@ -3,251 +3,172 @@ import { Flags } from '@oclif/core';
 import {
   AppleDeviceQuery,
   AppleDeviceQueryResult,
-  AppleDevicesByTeamIdentifierQueryResult,
 } from '../../credentials/ios/api/graphql/queries/AppleDeviceQuery';
-import { AppleTeamQuery } from '../../credentials/ios/api/graphql/queries/AppleTeamQuery';
 import { authenticateAsync, getRequestContext } from '../../credentials/ios/appstore/authenticate';
+import {
+  selectAppleDeviceOnAppleTeamAsync,
+  selectAppleTeamOnAccountAsync,
+} from '../../devices/queries';
 import formatDevice from '../../devices/utils/formatDevice';
 import { AppleDevice, Maybe } from '../../graphql/generated';
 import Log from '../../log';
 import { ora } from '../../ora';
 import { getOwnerAccountForProjectIdAsync } from '../../project/projectUtils';
-import { promptAsync, toggleConfirmAsync } from '../../prompts';
+import { toggleConfirmAsync } from '../../prompts';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
 
 export default class DeviceDelete extends EasCommand {
   static override description = 'remove a registered device from your account';
 
   static override flags = {
     'apple-team-id': Flags.string(),
-    udid: Flags.string({ multiple: true }),
+    udid: Flags.string({ multiple: false, description: 'The device ID to disable' }),
+    ...EasPaginatedQueryFlags,
+    ...EasNonInteractiveAndJsonFlags,
   };
 
   static override contextDefinition = {
     ...EASCommandProjectConfigContext,
   };
 
   async runAsync(): Promise<void> {
-    let {
-      flags: { 'apple-team-id': appleTeamIdentifier, udid: udids },
-    } = await this.parse(DeviceDelete);
+    const { flags } = await this.parse(DeviceDelete);
+    const paginatedQueryOptions = getPaginatedQueryOptions(flags);
+    let { 'apple-team-id': appleTeamIdentifier, udid } = flags;
     const {
       projectConfig: { projectId },
     } = await this.getContextAsync(DeviceDelete, {
-      nonInteractive: false,
+      nonInteractive: paginatedQueryOptions.nonInteractive,
     });
-
     const account = await getOwnerAccountForProjectIdAsync(projectId);
+    let appleTeamName;
+
+    if (paginatedQueryOptions.json) {
+      enableJsonOutput();
+    }
 
     if (!appleTeamIdentifier) {
-      appleTeamIdentifier = await this.askForAppleTeamAsync(account.name);
+      const appleTeam = await selectAppleTeamOnAccountAsync({
+        accountName: account.name,
+        selectionPromptTitle: `What Apple Team would you like to list devices for?`,
+        paginatedQueryOptions,
+      });
+      appleTeamIdentifier = appleTeam.id;
+      appleTeamName = appleTeam.appleTeamName;
     }
 
     assert(appleTeamIdentifier, 'No team identifier is specified');
 
-    const appleDevicesResult = await this.getDevicesForTeamAsync(account.name, appleTeamIdentifier);
-
-    if (!appleDevicesResult) {
-      return;
-    }
-
-    const { appleTeamName, appleDevices } = appleDevicesResult;
+    const chosenDevice = udid
```

Comment:
Just making a note here that we are dropping support for deleting multiple devices in a single command run. 

Based on my discussion in and beyond #1385, we're choosing to drop support for multi-select prompts + pagination. The idea is that we'd prefer a better UX if we were to support it and we don't have the bandwidth to do that just now. So we'll remove the ability and re-enable it at some later date.